src/libDirectoryService/PoWProcessing.cpp

```diff
@@ -117,8 +117,8 @@ bool DirectoryService::ProcessPoWPacketSubmission(
 
   LOG_GENERAL(INFO, "PoW solutions received in this packet: " << tmp.size());
   for (auto& sol : tmp) {
-    // No point processing the other solutions if DS Block consensus is starting
-    if ((m_state == DSBLOCK_CONSENSUS_PREP) || (m_state == DSBLOCK_CONSENSUS)) {
+    // No point processing other solutions if no more pow submission state.
+    if (m_state != POW_SUBMISSION) {
```

Comment:
There can be a timing issue. Let's say a DS node is at waiting to receive the final block at vacuous epoch before transiting to POW submission, it may drop POW submission from nodes that are faster than it